### DISCUSSION TITLE
Compiling EPOCH with gcc 8.1.0

### DISCUSSION THREAD

**Discussion by jakob-jonnerby** [epochpic/epoch#153] [Q&A]
Hello,

I am trying to install epoch2d on my university cluster and are running into an unexpected error. Apologies in advance if this is a simple mistake on my part.

I've recently compiled openmpi 4.1.0 using gcc 8.1.0, however when I try compiling epoch2d (version 4.3.7) using make COMPILER=gfortran I get the following error:

mpif90 -c -O3 -I/usr/include -Iobj -Jobj -o obj/sdf_common.o src/io/sdf_common.f90
f951: Fatal Error: Reading module ‘sdf_job_info’ at line 1 column 2: Unexpected EOF
compilation terminated.
make: *** [sdf_common.o] Error 1

Any suggestions are greatly appreciated!

**Comment by TomGoffrey** [comment]
Before investigating the compilation error, is "version 4.3.7" a typo? That's almost seven years old, and I would not recommend using it.

**Reply by jakob-jonnerby** [reply]
Thank you, yes I had missed that I wasn't using the latest version. It managed to compile now, but am still running into problems when trying to run one of the example input decks (ionisation.deck). I get the following message (still using gcc 8.1.0 and openmpi 4.1.0):

 Welcome to EPOCH2D version 4.17.15   (commit v4.17.15-0-gbbc1c3fa-clean)

` *************************************************************
 The code was compiled with no compile time options
 *************************************************************
 Code is running on 1 processing elements

 Specify output directory
At line 86 of file src/epoch2d.F90 (unit = 5, file = 'stdin')
Fortran runtime error: End of file

Error termination. Backtrace:
#0  0x7f96cb180be9 in read_sf
	at ../.././libgfortran/io/transfer.c:395
#1  0x7f96cb17dae6 in read_default_char1
	at ../.././libgfortran/io/read.c:420
#2  0x7f96cb182104 in formatted_transfer_scalar_read
	at ../.././libgfortran/io/transfer.c:1591
#3  0x7f96cb182f1c in formatted_transfer
	at ../.././libgfortran/io/transfer.c:2270
#4  0x494a6b in pic
	at src/epoch2d.F90:86
#5  0x40355c in main
	at src/epoch2d.F90:33
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------`
